core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -379,6 +371,8 @@ public Object get(int i) {
         return upperBounds;
       case 13:
         return keyMetadata;
+      case 14:
```

Comment:
The `size` method also needs to be updated.

Reply:
Ah, good catch.